src/inspect/inspector.rs

```diff
@@ -42,24 +45,40 @@ impl Inspector {
                 debug_info,
                 _non_exhaustive: (),
             }) => {
-                let symbols = if *debug_info {
+                let backend = if *debug_info {
                     let debug_line_info = true;
                     let debug_info_symbols = true;
-                    let resolver = DwarfResolver::open(path, debug_line_info, debug_info_symbols)?;
-
-                    names
-                        .iter()
-                        .map(|name| resolver.find_addr(name, &opts).ok().unwrap_or_default())
-                        .collect()
+                    let dwarf = DwarfResolver::open(path, debug_line_info, debug_info_symbols)?;
+                    let backend = ElfBackend::Dwarf(Rc::new(dwarf));
+                    backend
                 } else {
-                    let resolver = ElfResolver::new(path)?;
-                    names
-                        .iter()
-                        .map(|name| resolver.find_addr(name, &opts).unwrap_or_default())
-                        .collect()
+                    let elf = ElfParser::open(path)?;
+                    let backend = ElfBackend::Elf(Rc::new(elf));
+                    backend
                 };
 
-                Ok(symbols)
+                let resolver = ElfResolver::with_backend(path, 0, backend)?;
+                let syms = names
+                    .iter()
+                    .map(|name| {
+                        let mut syms = resolver.find_addr(name, &opts).unwrap_or_default();
+                        let () = syms.iter_mut().for_each(|sym| {
+                            if opts.offset_in_file {
+                                if let Some(off) = resolver.addr_file_off(sym.address) {
+                                    sym.file_offset = off;
```

Comment:
just thinking out loud: should we always calculate file_offset and not even have an `opts.offset_in_file`?

Reply:
I suppose we could. Will check how expensive `addr_file_off` is and whether configurability has merrit. But let's do that as a follow-on, as the change at hand is effectively only a code "move"/copy.